src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -86,26 +83,57 @@ struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Struc
      *
      * @note If the locationName is empty string and aFloorNumber and aAreaTypeTag are null, locationInfo will be set to null.
      */
-    LocationStructureWrapper & operator=(const LocationStructureWrapper & aOther)
+    AreaStructureWrapper & operator=(const AreaStructureWrapper & aOther)
     {
-        if (aOther.locationInfo.locationInfo.IsNull())
+        areaID = aOther.areaID;
+        mapID  = aOther.mapID;
+
+        if (aOther.areaDesc.locationInfo.IsNull())
+        {
+            areaDesc.locationInfo.SetNull();
+        }
+        else
+        {
+            areaDesc.locationInfo.SetNonNull();
+
+            // deep copy the name.
+            auto sizeToCopy = sizeof(mLocationNameBuffer);
+            if (aOther.areaDesc.locationInfo.Value().locationName.size() < sizeToCopy)
+            {
+                sizeToCopy = aOther.areaDesc.locationInfo.Value().locationName.size();
```

Comment:
Is it allowed to use STL algos in the server code? If yes, consider using std::min here, rather than the if statement above and this statement.

Reply:
See [here](https://github.com/project-chip/connectedhomeip/pull/34738/commits/b73083277a6df686bdcc6f071dfc4835f586ea0a).